sycl/include/sycl/ext/oneapi/experimental/raw_kernel_arg.hpp

```diff
@@ -27,6 +27,8 @@ class raw_kernel_arg {
   size_t MArgSize;
 
   friend class sycl::handler;
+  // For sycl_ext_oneapi_graph integration
+  friend class detail::dynamic_parameter_impl;
```

Comment:
You did the original implementation of `raw_kernel_arg` @steffenlarsen, is using `friend` like this acceptable? I didn't want to introduce a PIMPL pattern for such a simple class or start exposing the member variables publicaly